optimade/models/links.py

```diff
@@ -39,11 +39,11 @@ class Aggregate(Enum):
 class LinksResourceAttributes(Attributes):
     """Links endpoint resource object attributes"""
 
-    name: str = StrictField(
+    name: Optional[str] = StrictField(
         ...,
         description="Human-readable name for the OPTIMADE API implementation, e.g., for use in clients to show the name to the end-user.",
     )
-    description: str = StrictField(
+    description: Optional[str] = StrictField(
```

Comment:
There should still be a "name"and a "description" key when the links object is initialized. So technically it still follows the spec. 
But setting them as Optional allows their value to be `null`.  This way, the values do not need to be read from the database when they are not specified in the response_fields parameter.
If you want we could change the spec so these fields are not allowed to be null.

Reply:
The models here basically define the spec though (in its OpenAPI manifestation, at least), so we should be mindful of that. I'm surprised that this change doesn't alter the OpenAPI schemas, but if it doesn't, then I guess there is no problem with this...

Reply:
Looking into this a bit more, are you trying to stop these fields from appearing in the `required` section of the schema, so that they do not get added by default to the database projection? Just setting the type hint to `Optional` will not do this, you also need to give them a default value of `None`.

Reply:
I am not trying to stop them from appearing in the `required` section of the schema. I tried to keep the schemas the same.
If I specify the type as `str` I get an error because I am passing the value `None`, in case the `response_fields` parameter is present, and it is not one of the `response_fields`, when I generate an instance of the class. I could also set the type to `Union[str, None]` if you prefer.  It seems the schema generator does not consider None as a value type, so the schema stays the same.

If I give a default value for example: `None`, you could create an instance of the class without specifying a value for that property, so it is considered optional. This is apparently the type optional that is used for the schema.